internal/controller/pipelinerollout_controller.go

```diff
@@ -391,11 +401,13 @@ func (r *PipelineRolloutReconciler) processExistingPipeline(ctx context.Context,
 			return err
 		}
 		pipelineRollout.Status.MarkDeployed(pipelineRollout.Generation)
+		r.customMetrics.ReconciliationDuration.WithLabelValues(ControllerPipelineRollout, "update").Observe(time.Since(syncStartTime).Seconds())
```

Comment:
I don't think we know if this was an actual update or a "no-op"

Reply:
`if pipelineNeedsToUpdate {...}`

Reply:
this seems good